File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/flyweight/table/TableCell.java
// TableCell.java
package com.bigboxcode.designpattern.flyweight.table;

public class TableCell {
    private int width;
    private String text;

    public TableCell(int width) {
        this.width = width;
    }

    public void setText(String text) {
        this.text = text;
    }

    public void draw() {
       System.out.println("Drawing cell : width = " + width + " | text = " + text);
    }
}

File: design-patterns/php/flyweight/tablecell/TableCell.php
<?php
// TableCell.php

namespace BigBoxCode\DesignPattern\Flyweight\TableCell;


class TableCell {
    private int $width;
    private string $text = '';

    public function __construct(int $width) {
        $this->width = $width;
    }

    public function setText(string $text) {
        $this->text = $text;
    }

    public function draw(): void {
        echo "Drawing cell : width = " . $this->width . " | text = " . $this->text . "\n";
    }
}